compiler/z/codegen/OMRRegisterDependency.hpp

```diff
@@ -410,7 +410,7 @@ class RegisterDependencyConditions: public OMR::RegisterDependencyConditions
          // Printf added so it triggers some output even in prod build.
          // If this failure is triggered in a prod build, you might
          // not get a SEGV nor any meaningful error msg.
-         TR_ASSERT(0,"ERROR: addPostCondition list overflow\n");
+         TR_ASSERT_FATAL(0,"ERROR: addPostCondition list overflow\n");
          _cg->comp()->failCompilation<TR::CompilationException>("addPostCondition list overflow, abort compilation\n");
```

Comment:
The extra `failCompilation` is no longer needed since `TR_ASSERT_FATAL` terminates the process.
This whole if statement can be simplified to

```
TR_ASSERT_FATAL(_addCursorForPost < _numPostConditions, "addPostCondition list overflow");
```

(The extra "ERROR" at the start of the message adds no value since an assert failure is obviously an error).

Reply:
Excellent point Leo. Further to that, let's think about what would happen if we hit this assert in the wild. Would there be any other information that we could print that would be helpful in this case? I'd imagine the first thing one may want to know when debugging such a problem is what is the value of `_numPostConditions`? It would be helpful to print that as part of the assert, along with any other information you may think would be useful.

Reply:
Besides `_numPostConditions` and `_addCursorForPost`, is there anything else that could be useful while debugging? I was just printing those two when debugging a bug in `inlineVectorizedStringIndexOf` Could register name be helpful?

Reply:
Yes, the virtual register pointer and name would be very useful, in addition to `_numPostConditions`.